rpcs3/Emu/System.cpp

```diff
@@ -243,6 +243,39 @@ void fixup_ppu_settings()
 	}
 }
 
+void dump_executable(std::span<const u8> data, main_ppu_module* _main, std::string_view title_id)
+{
+	// Format filename and directory name
+	// Make each directory for each file so tools like IDA can work on it cleanly
+	const std::string dir_path = fs::get_cache_dir() + "ppu_progs/" + std::string{!title_id.empty() ? title_id : "untitled"} + fmt::format("-%s-%s", fmt::base57(_main->sha1), _main->path.substr(_main->path.find_last_of('/') + 1))  + '/';
+	const std::string filename = dir_path + "exec.elf";
+
+	if (fs::create_dir(dir_path) || fs::g_tls_error == fs::error::exist)
+	{
+		if (fs::file out{filename, fs::create + fs::write})
+		{
+			if (out.size() == data.size())
+			{
+				// Risky optimization: assume if file size match they are equal and does not need to rewrite it
```

Comment:
Why does this need such an optimization if it's a debug tool anyway?

Reply:
So it would convenient to use it most of the time, the risks are extremely low and depend on disk errors. The write itself is not atomic so it's not perfect anyways.

Reply:
And even if there are errors, common sensse says to delete the file and redump it.